lib/api/apiUtils/object/coldStorage.js

```diff
@@ -28,6 +33,99 @@ function getAmzRestoreResHeader(objMD) {
     return undefined;
 }
 
+
+/**
+ * Check if restore can be done.
+ *
+ * @param {BucketInfo} bucketMD - bucket instance
+ * @param {ObjectMD} objectMD - object metadata
+ * @return {ArsenalError|boolean} - undefined if the conditions for RestoreObject are fulfilled
+ */
+function _validateStartRestore(bucketMD, objectMD) {
+    const isLocationCold = locationConstraints[bucketMD.getLocationConstraint()]?.isCold;
```

Comment:
We should not be able to create a bucket in a cold storage location. Would this always be false?

Reply:
I think the archive process will update the locationConstraint of bucketMD to a cold storage location, so that this should be true after object being archived